pkg/reconciler/revision/revision.go

```diff
@@ -46,7 +44,8 @@ import (
 const digestResolutionTimeout = 60 * time.Second
 
 type resolver interface {
-	Resolve(context.Context, string, k8schain.Options, sets.String) (string, error)
+	Resolve(*v1.Revision, k8schain.Options, sets.String, time.Duration) ([]v1.ContainerStatus, error)
```

Comment:
Should we move the timeout out of the signature and into the constructor?

Reply:
we could do, but I think we still eventually want to allow the timeout to be set in the configmap which is easier if this is done in the interface

Reply:
I think it'd end up being a similar signature change since we'd probably extract the config from context, but we can change this interface whenever since it's private.